pkg/odo/cli/component/create.go

```diff
@@ -475,48 +470,49 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			}
 		}
 
-		// Set devfileMetadata struct
-		co.devfileMetadata.componentType = componentType
-		co.devfileMetadata.componentName = strings.ToLower(componentName)
-		co.devfileMetadata.componentNamespace = strings.ToLower(componentNamespace)
+		if isDevfileRegistryPresent {
```

Comment:
This `isDevfileRegistryPresent` flag should only apply to the case where creating devfile component from registry, which means if the registry list is empty user still can use the existing devfile to create devfile component. So `isDevfileRegistryPresent` flag should be used here https://github.com/openshift/odo/pull/3291/files#diff-58fe314b4e3f9120402717ab8c2b3049L512 instead of the whole code block.

Reply:
actually it doesnt matter because `isDevfileRegistryPresent` is true by default but i moved it because local devfile is not devfile registry. So later when ppl comes to read the code they dont get confused,

But I had to move the header `Validation` to inside the if blocks to keep interactive mode clean.